test/unit/org/apache/cassandra/sensors/SensorsIndexWriteTest.java

```diff
@@ -0,0 +1,254 @@
+
+    @Test
+    public void testSingleRowMutationWithSecondaryIndex()
+    {
+        ColumnFamilyStore standardStore = SensorsTestUtil.discardSSTables(KEYSPACE1, CF_STANDARD);
+        Context standardContext = new Context(KEYSPACE1, CF_STANDARD, standardStore.metadata.id.toString());
+
+        ColumnFamilyStore secondaryIndexStore = SensorsTestUtil.discardSSTables(KEYSPACE1, CF_STANDARD_SECONDARY_INDEX);
+        Context secondaryIndexContext = new Context(KEYSPACE1, CF_STANDARD_SECONDARY_INDEX, secondaryIndexStore.metadata.id.toString());
+
+        String partitionKey = "0";
+        Mutation standardMutation = new RowUpdateBuilder(standardStore.metadata(), 0, partitionKey)
+                                    .add("val", "hi there")
+                                    .build();
+        handleMutation(standardMutation);
+
+        Sensor standardSensor = SensorsTestUtil.getThreadLocalRequestSensor(standardContext, Type.WRITE_BYTES);
+        assertThat(standardSensor.getValue()).isGreaterThan(0);
+        Sensor standardRegistrySensor = SensorsTestUtil.getRegistrySensor(standardContext, Type.WRITE_BYTES);
+        assertThat(standardRegistrySensor).isEqualTo(standardSensor);
+
+        // check global registry is synchronized for Standard table
+        assertThat(standardRegistrySensor.getValue()).isEqualTo(standardSensor.getValue());
+        assertWriteResponseSensors(standardSensor.getValue(), standardRegistrySensor.getValue(), CF_STANDARD);
+
+        Mutation secondaryIndexMutation = new RowUpdateBuilder(secondaryIndexStore.metadata(), 0, partitionKey)
+                                          .add("val", "hi there")
+                                          .build();
+        handleMutation(secondaryIndexMutation);
+
+        Sensor secondaryIndexSensor = SensorsTestUtil.getThreadLocalRequestSensor(secondaryIndexContext, Type.INDEX_WRITE_BYTES);
+        // Writing the same amount of data to an indexed column should generate at least the same number of bytes (the SecondaryIndex write bytes >= the vanilla write bytes)
+        assertThat(secondaryIndexSensor.getValue()).isGreaterThanOrEqualTo(standardSensor.getValue());
+        Sensor secondartyIndexRegistrySensor = SensorsTestUtil.getRegistrySensor(secondaryIndexContext, Type.INDEX_WRITE_BYTES);
+        assertThat(secondartyIndexRegistrySensor).isEqualTo(secondaryIndexSensor);
+
+        // check global registry is synchronized for Secondary Index table
+        assertThat(secondartyIndexRegistrySensor.getValue()).isEqualTo(secondaryIndexSensor.getValue());
+        assertIndexResponseSensors(secondaryIndexSensor.getValue(), secondartyIndexRegistrySensor.getValue(), CF_STANDARD_SECONDARY_INDEX);
+    }
+
+    private static void handleMutation(Mutation mutation)
+    {
+        MutationVerbHandler.instance.doVerb(Message.builder(Verb.MUTATION_REQ, mutation).build());
+    }
+
+    private void assertWriteResponseSensors(double requestValue, double registryValue, String table)
```

Comment:
Looks like the only things changing between this method and the following are the params names (write vs index). Can we refactor them and pass those names as parameters?